tests/tests_unit/test_api/test_functions.py

```diff
@@ -0,0 +1,1061 @@
+import os
+from unittest.mock import patch
+
+import pytest
+from requests import PreparedRequest
+
+from cognite.client import CogniteClient
+from cognite.client._api.functions import (
+    _extract_requirements_from_doc_string,
+    _extract_requirements_from_file,
+    _get_requirements_handle,
+    _using_client_credential_flow,
+    _validate_requirements,
+    validate_function_folder,
+)
+from cognite.client.data_classes import (
+    Function,
+    FunctionCall,
+    FunctionCallList,
+    FunctionCallLog,
+    FunctionList,
+    FunctionSchedule,
+    FunctionSchedulesList,
+    FunctionsLimits,
+)
+from cognite.client.data_classes.functions import FunctionsStatus
+from cognite.client.exceptions import CogniteAPIError
+from tests.utils import jsgz_load
+
+
+def post_body_matcher(params):
+    """
+    Used for verifying post-bodies to mocked endpoints. See the `match`-argument in `rsps.add()`.
+    """
+
+    def match(request_body):
+        if request_body is None:
+            return params is None, None
+        else:
+            if isinstance(request_body, PreparedRequest):
+                decompressed_body = jsgz_load(request_body.body)
+            else:
+                decompressed_body = jsgz_load(request_body)
+            sorted_params = sorted(params.items())
+            sorted_body = sorted(decompressed_body.items())
+
+            res = sorted_params == sorted_body
+            return res, None
+
+    return match
+
+
+COGNITE_CLIENT = CogniteClient()
```

Comment:
Use the cognite_client fixture instead, or this will fail when running tests with the interactive auth flow.